pkg/provision/engine.go

```diff
@@ -81,6 +87,23 @@ func (e *Engine) Run(ctx context.Context) error {
 
 	cReservation := e.source.Reservations(ctx)
 
+	// Indicates wether provisiond cleaned up resources after
+	// booting and provisioning all reservations
+	triggeredCleanupOnBoot := false
```

Comment:
I would actually have put this into the source.go file instead of here. But I guess it doesn't really matter.